examples/sampler.cpp

```diff
@@ -0,0 +1,404 @@
+        {
+            std::stringstream coarsename;
+            coarsename << "coarse_" << sample;
+            SaveFigure(sol_upscaled, ufespace, coarsename.str());
+            std::stringstream finename;
+            finename << "fine_" << sample;
+            SaveFigure(sol_fine, ufespace, finename.str());
+        }
+
+        if (myid == 0)
+        {
+            std::cout << "  Sample " << sample << ":" << std::endl;
+            std::cout << "    fine: iterations: " << fine_iterations
+                      << ", time: " << fine_time << std::endl;
+            std::cout << "    coarse: iterations: " << coarse_iterations
+                      << ", time: " << coarse_time << std::endl;
+            std::cout << "    p_error: " << finest_p_error << std::endl;
+        }
+    }
+
+    double count = static_cast<double>(num_samples);
+    if (count > 1.1)
+    {
+        m2_upscaled *= (1.0 / (count - 1.0));
+        m2_fine *= (1.0 / (count - 1.0));
+    }
+
+    serialize["total-coarse-iterations"] = picojson::value((double) total_coarse_iterations);
+    serialize["total-fine-iterations"] = picojson::value((double) total_fine_iterations);
+    serialize["fine-mean-typical"] = picojson::value(
+                                         mean_fine[mean_fine.Size() / 2]);
+    serialize["fine-mean-l1"] = picojson::value(
+                                    mean_fine.Norml1() / static_cast<double>(mean_fine.Size()));
+    serialize["coarse-mean-l1"] = picojson::value(
+                                      mean_upscaled.Norml1() / static_cast<double>(mean_upscaled.Size()));
+    serialize["coarse-mean-typical"] = picojson::value(
+                                           mean_upscaled[mean_upscaled.Size() / 2]);
+    serialize["fine-variance-mean"] = picojson::value(
+                                          m2_fine.Sum() / static_cast<double>(m2_fine.Size()));
+    serialize["coarse-variance-mean"] = picojson::value(
+                                            m2_upscaled.Sum() / static_cast<double>(m2_upscaled.Size()));
+    serialize["max-p-error"] = picojson::value(max_p_error);
+
+    if (visualization)
+    {
+        Visualize(mean_upscaled, ufespace, 1);
+        Visualize(mean_fine, ufespace, 0);
+        if (count > 1.1)
+        {
+            Visualize(m2_upscaled, ufespace, 11);
+            Visualize(m2_fine, ufespace, 10);
+        }
+    }
+    if (save_statistics)
+    {
+        std::stringstream coarsename;
+        coarsename << "coarse_mean";
+        SaveFigure(mean_upscaled, ufespace, coarsename.str());
+        std::stringstream finename;
+        finename << "fine_mean";
+        SaveFigure(mean_fine, ufespace, finename.str());
+        std::stringstream coarsenamev;
+        coarsenamev << "coarse_variance";
+        SaveFigure(mean_upscaled, ufespace, coarsenamev.str());
+        std::stringstream finenamev;
+        finenamev << "fine_variance";
+        SaveFigure(mean_fine, ufespace, finenamev.str());
```

Comment:
`mean_upscaled` and `mean_fine` are both saved twice.  Is `coarse_variance` supposed to save `m2_upscaled` (and fine equivalent)?

These can be done a bit cleaner w/o stringstream
```c++
{
        SaveFigure(mean_upscaled, ufespace, "coarse_mean");
        SaveFigure(mean_fine, ufespace, "fine_mean");
        SaveFigure(m2_upscaled, ufespace, "coarse_variance");
        SaveFigure(m2_fine, ufespace, "fine_variance");
}
```